src/js/tests.rs

```diff
@@ -0,0 +1,540 @@
+use super::*;
+use crate::build::{Origin, Module};
+use crate::type_;
+
+macro_rules! assert_js {
+    ($src:expr, $erl:expr $(,)?) => {
+        // println!("\n\n\n{}\n", $src);
+        let (mut ast, _) = crate::parse::parse_module($src).expect("syntax error");
+        ast.name = vec!["the_app".to_string()];
+        let mut modules = std::collections::HashMap::new();
+        // Probably a nicer way to do types values accessors
+        modules.insert(
+            "rocket_ship".to_string(), 
+            (Origin::Src, type_::Module { name: vec!["rocket_ship".to_string()], types: std::collections::HashMap::new(), values: std::collections::HashMap::new(), accessors: std::collections::HashMap::new()})
+        );
+        let ast =
+            crate::type_::infer_module(&mut 0, ast, &modules, &mut vec![])
+                .expect("should successfully infer");
+        let mut output = String::new();
+        let line_numbers = LineNumbers::new($src);
+        module(&ast, &line_numbers, &mut output).unwrap();
+        assert_eq!(($src, output), ($src, $erl.to_string()));
+    };
+}
+
+#[test]
+fn string_literals() {
+    assert_js!(
+        r#"
+fn go() {
+    "Hello, Gleam!"
+}
+"#,
+r#"function go() {
+    return "Hello, Gleam!"
```

Comment:
Let's use semicolons, otherwise we'd need to be clever and detect when code is being generated for which semicolons are mandatory.

Reply:
Done,